test/commitment/proof_of_work.cpp

```diff
@@ -0,0 +1,98 @@
+//---------------------------------------------------------------------------//
+// Copyright (c) 2024 Dmitrii Tabalin <d.tabalin@nil.foundation>
+//
+// MIT License
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+//---------------------------------------------------------------------------//
+
+#define BOOST_TEST_MODULE proof_of_work_test
+
+#include <boost/test/unit_test.hpp>
+
+#include <nil/crypto3/algebra/curves/pallas.hpp>
+#include <nil/crypto3/algebra/fields/pallas/scalar_field.hpp>
+#include <nil/crypto3/algebra/fields/pallas/base_field.hpp>
+
+#include <nil/crypto3/zk/commitments/detail/polynomial/proof_of_work.hpp>
+
+#include <nil/crypto3/hash/poseidon.hpp>
+#include <nil/crypto3/hash/detail/poseidon/poseidon_policy.hpp>
+#include <nil/crypto3/hash/keccak.hpp>
+
+#include <nil/crypto3/zk/transcript/fiat_shamir.hpp>
+
+using namespace nil::crypto3::algebra;
+using namespace nil::crypto3::zk::commitments;
+
+BOOST_AUTO_TEST_SUITE(proof_of_knowledge_test_suite)
+
+BOOST_AUTO_TEST_CASE(pow_poseidon_basic_test) {
+    using curve_type = curves::pallas;
+    using field_type = curve_type::base_field_type;
+    using integral_type = typename field_type::integral_type;
+    using policy = nil::crypto3::hashes::detail::mina_poseidon_policy<field_type>;
+    using poseidon = nil::crypto3::hashes::poseidon<policy>;
+    const std::uint64_t mask = 0x1F;
```

Comment:
Let's change mask value in our test and mask default value. It works properly, but it's a bit non-intuitive. For this example we'll have:

```
RESULT 210d06cf a0ccab10 05db754a f50be4d2 92038406 0f2c7d96 59971d1f b3dff6a7
  mask                 f 80000000 00000000 00000000 00000000 00000000 00000000
```
You see, it don't test first bits there, and we can't check it by range_check blueprint component.
Let's take something like this:
```
const std::uint64_t mask = 0xFFFF000000000000;
```